From 7526709d462b1c88777507261804f4c6aebf0048 Mon Sep 17 00:00:00 2001
From: Jason Hung <jhung@globalscaletechnologies.com>
Date: Mon, 12 Aug 2024 10:24:21 +0800
Subject: [PATCH 2/2] ax88179_programmer: fix compiler build warning

---
 ax88179_programmer.c | 14 ++++++++++----
 1 file changed, 10 insertions(+), 4 deletions(-)

diff --git a/ax88179_programmer.c b/ax88179_programmer.c
index 4990918..83b7d0f 100755
--- a/ax88179_programmer.c
+++ b/ax88179_programmer.c
@@ -322,8 +322,8 @@ static int writeeeprom_func(struct ax_command_info *info)
 
 	for (i = 0; i < wLen / 8; i++) {
 		int j = 8 * i;
-
-		fscanf(pFile, "%04X %04X %04X %04X %04X %04X %04X %04X%c",
+		int ret;
+		ret = fscanf(pFile, "%04X %04X %04X %04X %04X %04X %04X %04X%c",
 				(unsigned int *)&buf[j + 0],
 				(unsigned int *)&buf[j + 1],
 				(unsigned int *)&buf[j + 2],
@@ -332,6 +332,9 @@ static int writeeeprom_func(struct ax_command_info *info)
 				(unsigned int *)&buf[j + 5],
 				(unsigned int *)&buf[j + 6],
 				(unsigned int *)&buf[j + 7], c);
+			if (ret != 9) {
+				printf("error reading from file\n");
+			}
 	}
 
 	ioctl_cmd.ioctl_cmd = info->ioctl_cmd;
@@ -438,8 +441,8 @@ static int chgmac_func(struct ax_command_info *info)
 
 		for (i = 0; i < wLen / 8; i++) {
 			int j = 8 * i;
-
-			fscanf(pFile, "%04X %04X %04X %04X %04X %04X %04X %04X%c",
+			int ret;
+			ret = fscanf(pFile, "%04X %04X %04X %04X %04X %04X %04X %04X%c",
 					(unsigned int *)&buf[j + 0],
 					(unsigned int *)&buf[j + 1],
 					(unsigned int *)&buf[j + 2],
@@ -448,6 +451,9 @@ static int chgmac_func(struct ax_command_info *info)
 					(unsigned int *)&buf[j + 5],
 					(unsigned int *)&buf[j + 6],
 					(unsigned int *)&buf[j + 7], c);
+			if (ret != 9) {
+				printf("error reading from file\n");
+			}
 		}
 	} else {
 		ioctl_cmd.ioctl_cmd = AX88179_READ_EEPROM;
-- 
2.47.1

